lib/eclipse/include/opm/parser/eclipse/EclipseState/AquiferCT.hpp

```diff
@@ -0,0 +1,95 @@
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_AQUIFERCT_HPP
+#define OPM_AQUIFERCT_HPP
+
+/*
+  The AquiferCT which stands for AquiferCarterTracy is a data container object meant to hold the data for the aquifer carter tracy model.
+  This includes the logic for parsing as well as the associated tables. It is meant to be used by opm-grid and opm-simulators in order to
+  implement the Carter Tracy analytical aquifer model in OPM Flow.
+*/
+
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/A.hpp>
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Deck/DeckItem.hpp>
+#include <opm/parser/eclipse/Deck/DeckRecord.hpp>
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/Aqudims.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/TableContainer.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/AqutabTable.hpp>
+#include <boost/concept_check.hpp>
+
+namespace Opm {
+
+    class AquiferCT {
+        public:
+
+            struct AQUCT_data{
+
+                    // Aquifer ID
+                    int aquiferID;
+                    // Table IDs
+                    int inftableID, pvttableID;
+                    std::vector<int> cell_id;
+                    // Variables constants
+                    double  phi_aq , //aquifer porosity
+                            d0,   //aquifer datum depth
+                            C_t , //total compressibility
+                            r_o , //aquifer inner radius
+                            k_a , //aquifer permeability
+                            c1, // 0.008527 (METRIC, PVT-M); 0.006328 (FIELD); 3.6 (LAB)
+                            h , //aquifer thickness
+                            theta , //angle subtended by the aquifer boundary
+                            c2 ; //6.283 (METRIC, PVT-M); 1.1191 (FIELD); 6.283 (LAB).
+
+                    std::vector<double> td, pi;
+            };
+
+            AquiferCT(const EclipseState& eclState, const Deck& deck);
+
+            const std::vector<AquiferCT::AQUCT_data>& getAquifers() const;
+            const int getAqInflTabID(size_t aquiferIndex);
+            const int getAqPvtTabID(size_t aquiferIndex);
+    
+        private:
+  
+            std::vector<AquiferCT::AQUCT_data> m_aquct;
+    
+            inline void set_default_tables(std::vector<double>& td, std::vector<double>& pi)
+            {
+                std::vector<double> default_pressure_ = { 0.112, 0.229, 0.315, 0.376, 0.424, 0.469, 0.503, 0.564, 0.616, 0.659, 0.702, 0.735, 
```

Comment:
Is this the table from van Everdingen and Hurst that is also used by default in Eclipse?

Reply:
Yes, it is the constant terminal rate case for an infinite aquifer as given by Van Everdingen and Hurst
[https://www.onepetro.org/download/journal-paper/SPE-949305-G?id=journal-paper%2FSPE-949305-G]